src/Matrix.cpp

```diff
@@ -0,0 +1,181 @@
+	{
+		std::cout << "[" << matrix_to_print[i] << "]";
+		if ((i + 1) % matrix_to_print.p_columns == 0)
+		{
+			std::cout << std::endl;
+		}
+	}
+}
+
+
+//prints pixels in matrix,		[0.4213] [0.4213] [0.4213]			[0.4213] [0.4213] [0.4213] 
+//2x4 example, 8 pixels total	[0.4213] [0.4213] [0.4213]			[0.4213] [0.4213] [0.4213]
+matrix matrix::operator+ (const matrix& other_matrix) const
+{	
+	if (other_matrix.size == this->size && other_matrix.m_columns == this->m_columns && other_matrix.m_rows == this->m_rows)
+	{	
+		std::vector<double> new_values;
+
+		for (int i = 0; i < other_matrix.size; i++)
+		{
+			double added_value = other_matrix[i] + (*this)[i];
+			new_values.push_back(added_value);
+		}
+
+		matrix matrix_to_return(other_matrix.m_rows, other_matrix.m_columns, new_values);
+		return matrix_to_return;
+
+	}
+		std::cout << "Left Matrix: " << this->m_size << " Right matrix: " << other_matrix.m_size << std::endl;
+		throw std::invalid_argument("Cannot add matrices of different sizes.");
+}
+
+
+matrix matrix::operator-(const matrix& other_matrix)
+{
+	if (other_matrix.size == this->size && other_matrix.m_columns == this->m_columns && other_matrix.m_rows == this->m_rows)
+	{
+		std::vector<double> new_values;
+
+		for (int i = 0; i < other_matrix.size; i++)
+		{
+			double subtracted_value = (*this)[i] - other_matrix[i];
+			new_values.push_back(subtracted_value);
+		}
+
+		matrix matrix_to_return(other_matrix.m_rows, other_matrix.m_columns, new_values);
+		return matrix_to_return;
+
+	}
+	std::cout << "Left Matrix: " << this->m_size << " Right matrix: " << other_matrix.m_size << std::endl;
+	throw std::invalid_argument("Cannot subtract matrices of different sizes.");
+}
+
+matrix matrix::operator*(const matrix& other_matrix)
+{
+	if (this->m_columns == other_matrix.m_rows) 
+	{
+		std::vector<double> new_values;
+		int rows_of_new_matrix = this->m_rows;
+		int columns_of_new_matrix = m_columns;
+		int new_matrix_size = rows_of_new_matrix * columns_of_new_matrix;
+
+		for (int i = 0; i < new_matrix_size; i++) 
+		{
+
+		}
+
+		//fill this matrix out with right constructor
+		matrix matrix_to_return;
+		return matrix_to_return;
+	}
+	std::cout << "Left matrix columns: " << this->m_columns << "Right matrix rows: " << other_matrix.m_rows;
+	throw std::invalid_argument("Columns of left matrix must = rows of right matrix.");
+}
+
+//all this reference means is that the type at this address is *presumed* to be the type we give it.
+//the pointer is still just an address to a specific byte.
+double& matrix::operator[](const int index) const
+{
+	
+	if (index >= m_size || index < 0)
+	{
+		std::cout << "Matrix index: [" << index << "] is out of bounds. Size of matrix: " << m_size << std::endl;
+		throw std::invalid_argument("");
+	}
+	return m_ptr_double[index];
+}
+
+double& matrix::operator() (const int row, const int col) const
+{
+	if (row > 0 && col > 0 && col <= m_columns && row <= m_rows)
+	{
+		int row_of_element = row - 1;
```

Comment:
I'm confused. Why are we adjusting the row and column index that we specify when calling this?

I think that as we are both programmers, it would be more intuitive for us the refer to the first element as at index (0, 0).

Reply:
70d1b772772855d147f5fc50ae4e26de80df28fc

changed so element 0 is at (0,0) now